spec/src/main/asciidoc/architecture.adoc

```diff
@@ -282,11 +282,22 @@ how such application servers should behave if they want to support MicroProfile
 Metrics from all applications and scopes should be available under a single REST endpoint ending with `/metrics` similarly as
 in case of single-application deployments (microservices).
 
-To help distinguish between metrics pertaining to each deployed application, a tag named `mp_app` should be added to each metric. 
+To help distinguish between metrics pertaining to each deployed application, a tag named `mp_app` should be added to each metric.
 
-The value of the `mp_app` tag should be passed by the application server to the application via a MicroProfile Config property named `mp.metrics.appName`.
-It should be possible to override this value by bundling the file `META-INF/microprofile-config.properties` within the application archive
-and setting a custom value for the property `mp.metrics.appName` inside it.
+The value of the `mp_app` tag should be passed by the application server to the application via a MicroProfile Config property named `mp.metrics.appName`. The property is to be defined at the application level through the `META-INF/microprofile-config.properties` file within the application archive.
+
+Due to the restriction imposed on metrics of the same-name to contain the same set of tag names. A conflict for metric registration may occur for runtimes that support _vendor_ metrics that can originate from both a server context and an application context. For example, a vendor implementation that provides servlet metrics could track servlet requests to the application(s) and to the server. However, the servlet metrics that track requests to the application may originate from the application-context and may include the `mp_app` tag. This will cause a mismatched set of tags compared to the servlet metrics that originate from the server-context.
+
+To avoid this, it is recommended that the MicroProfile Config property `mp.metrics.defaultAppName` be defined in conjunction with the use of the `mp.metrics.appName` property. The `mp.metrics.defaultAppName` will be defined at the  server level and will provide a _default_ `mp_app` tag value for server level metrics.
```

Comment:
we need to clearly state what an impl must do with the mp.metrics.defaultAppName config value.

Reply:
Including the change specified here https://github.com/eclipse/microprofile-metrics/pull/777#discussion_r1294654272

The full text is modified to:
```
To avoid this, it is recommended that the MicroProfile Config property `mp.metrics.defaultAppName` be defined in conjunction with the use of the `mp.metrics.appName` property. The `mp.metrics.defaultAppName` should be defined by users at the server level. The value of the property will provide a _default_ `mp_app` tag value for server level metrics (i.e., metrics that do not originate from the application).
```

The last sentence should for this comment.